### PR TITLE
Add UnionShape for SparseTensor TypeProto

### PR DESCRIPTION
**Pull request by yuslepukhin**
This is a missing utility for SparseTensor type proto

### CODE REVIEW

File: onnx/defs/shape_inference.cc [hunk 1]
```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "shape_inference.h"
+
+namespace ONNX_NAMESPACE {
+
+void UnionShapeInfo(const TensorShapeProto& source_shape, TensorShapeProto& target_shape) {
```
Comment (gramalingam) [inline onnx/defs/shape_inference.cc:9]: Thanks for the refactoring and moving it into a .cc file. May be useful to add a comment above that source_shape and target_shape are assumed to have same dim_size().

**Comment by gramalingam** [comment]
LGTM, thanks! Just a minor comment above.

**Review by askhade — APPROVED** [review]
LGTM

File: onnx/defs/shape_inference.cc (hunk 1 above)
Comment (yuslepukhin) [reply onnx/defs/shape_inference.cc:9]: I will remove it from the header

**Review by gramalingam — APPROVED** [review]

**Pull request merged**